src/cmap/auth/mongo_credentials.ts

```diff
@@ -32,6 +33,10 @@ export interface AuthMechanismProperties extends Document {
   SERVICE_REALM?: string;
   CANONICALIZE_HOST_NAME?: GSSAPICanonicalizationValue;
   AWS_SESSION_TOKEN?: string;
+  DEVICE_NAME?: string;
+  PRINCIPAL_NAME?: string;
+  REQUEST_TOKEN_CALLBACK?: OIDCRequestFunction;
+  REFRESH_TOKEN_CALLBACK?: OIDCRefreshFunction;
```

Comment:
Perhaps a small bit of documentation would be nice here
```suggestion
  /** A name for the device workflow that is one of "aws", "azure", or "gcp". */
  DEVICE_NAME?: string;
  /** This is required by the server when more than one Identity Provider is configured */
  PRINCIPAL_NAME?: string;
  REQUEST_TOKEN_CALLBACK?: OIDCRequestFunction;
  REFRESH_TOKEN_CALLBACK?: OIDCRefreshFunction;
```

etc. more for our reference than anything. Also do we want to put 'aws' or the other cloud providers as the type for DEVICE_NAME?

Reply:
Added comments.